src/SatNav.cc

```diff
@@ -0,0 +1,226 @@
+  public: sdf::ElementPtr sdf{nullptr};
+};
+
+//////////////////////////////////////////////////
+SatNav::SatNav()
+  : dataPtr(new SatNavPrivate)
+{
+}
+
+//////////////////////////////////////////////////
+SatNav::~SatNav()
+{
+  delete this->dataPtr;
+  this->dataPtr = nullptr;
+}
+
+//////////////////////////////////////////////////
+SatNav::SatNav(const SatNav &_satnav)
+  : dataPtr(new SatNavPrivate(*_satnav.dataPtr))
+{
+}
+
+//////////////////////////////////////////////////
+SatNav::SatNav(SatNav &&_satnav) noexcept
+  : dataPtr(std::exchange(_satnav.dataPtr, nullptr))
+{
+}
+
+//////////////////////////////////////////////////
+SatNav &SatNav::operator=(const SatNav &_satnav)
+{
+  return *this = SatNav(_satnav);
+}
+
+//////////////////////////////////////////////////
+SatNav &SatNav::operator=(SatNav &&_satnav) noexcept
+{
+  std::swap(this->dataPtr, _satnav.dataPtr);
+  return * this;
+}
+
+//////////////////////////////////////////////////
+Errors SatNav::Load(ElementPtr _sdf)
+{
+  Errors errors;
+
+  this->dataPtr->sdf = _sdf;
+
+  // Check that sdf is a valid pointer
+  if (!_sdf)
+  {
+    errors.push_back({ErrorCode::ELEMENT_MISSING,
+        "Attempting to load SATNAV, but the provided SDF "
+        "element is null."});
+    return errors;
+  }
+
+  // Check that the provided SDF element is a <satnav> element.
+  // This is an error that cannot be recovered, so return an error.
+  if (_sdf->GetName() != "satnav" && _sdf->GetName() != "gps")
+  {
+    errors.push_back({ErrorCode::ELEMENT_INCORRECT_TYPE,
+        "Attempting to load SATNAV, but the provided SDF element is "
+        "not a <satnav>."});
+    return errors;
+  }
+
+  // Load satnav sensor properties
+  if (_sdf->HasElement("position_sensing"))
+  {
+    sdf::ElementPtr elem = _sdf->GetElement("position_sensing");
+    if (elem->HasElement("horizontal"))
+    {
+      sdf::ElementPtr horiz = elem->GetElement("horizontal");
+      if (horiz->HasElement("noise"))
+      {
+        this->dataPtr->horizontalPositionNoise.Load(horiz->GetElement("noise"));
+      }
+    }
+    if (elem->HasElement("vertical"))
+    {
+      sdf::ElementPtr vert = elem->GetElement("vertical");
+      if (vert->HasElement("noise"))
+      {
+        this->dataPtr->verticalPositionNoise.Load(vert->GetElement("noise"));
+      }
+    }
+  }
+  if (_sdf->HasElement("velocity_sensing"))
+  {
+    sdf::ElementPtr elem = _sdf->GetElement("velocity_sensing");
+    if (elem->HasElement("horizontal"))
+    {
+      sdf::ElementPtr horiz = elem->GetElement("horizontal");
+      if (horiz->HasElement("noise"))
+      {
+        this->dataPtr->horizontalVelocityNoise.Load(horiz->GetElement("noise"));
+      }
+    }
+    if (elem->HasElement("vertical"))
+    {
+      sdf::ElementPtr vert = elem->GetElement("vertical");
+      if (vert->HasElement("noise"))
+      {
+        this->dataPtr->verticalVelocityNoise.Load(vert->GetElement("noise"));
+      }
+    }
+  }
+
+
```

Comment:
Nit: can we leave only 1 blank line? :grimacing:

Reply:
https://github.com/osrf/sdformat/pull/453/commits/b128c9cc8c5aad02e8db480bd25626ae8769c545